docs/cep/tutorials/transforming-data.md

```diff
@@ -6,89 +6,81 @@ sidebar_position: 4
 
         ```sql
-        from TempStream
+        INSERT INTO OutputStream
+        FROM TempStream;
         ```
-
-    1. To group by a specific attribute (by the `roomNo` attribute in this example), specify it via the `group by` clause as shown below.
-
+    B. Add a `select` to select the fields required to calculate the average temperature. Apply the `avg()` to the `temp` attribute, and then specify `avgTemp` as the name with which the result should be output. 
+    
         ```sql
-        insert into OutputStream
-        select roomNo, avg(temp) as avgTemp
-        from TempStream
-        group by roomNo;
+        SELECT roomNo, avg(temp) AS avgTemp
         ```
 
-    1. Add the `insert into` clause with the name of the output stream to indicate that the processed events are directed to that stream.
+    C. Add the `from` clause with the name of the input stream to indicate that the events to be processed are taken from the input stream.
 
         ```sql
-        insert into OutputStream
-        from TempStream;
+        FROM TempStream
         ```
-      
-    1. To group by a specific attribute (by the `roomNo` attribute in this example), specify it via the `group by` clause as shown below.
+
+    D. To group by a specific attribute (by the `roomNo` attribute in this example), specify it via the `group by` clause as shown below.
 
         ```sql
-        insert into OutputStream
-        select roomNo, deviceID, avg(temp) as avgTemp
-        from TempStream
-        group by roomNo;
+        INSERT INTO OutputStream
+        SELECT roomNo, avg(temp) AS avgTemp
+        FROM TempStream
+        GROUP BY roomNo;
         ```
 
-1. Save the stream application. The completed stream application is as follows.
+9. Save the stream worker. The completed stream worker is as follows.
 
     ```sql
     @App:name("TemperatureApp")
     @App:description("Calculate an average temperature of the room")
     @App:qlVersion("2")
-
-	CREATE STREAM TempStream (deviceID long, roomNo int, temp double);
-
-	CREATE STREAM OutputStream WITH (type='stream', stream='OutputStream', map.type='json') (roomNo int, avgTemp double);
-
-    insert into OutputStream
-    select roomNo, avg(temp) as avgTemp
-    from TempStream
-    group by roomNo;
+    
+    CREATE STREAM TempStream (deviceID long, roomNo int, temp double);
+    
+    CREATE SINK OutputStream WITH (type='stream', stream='OutputStream', map.type='json') (roomNo int, avgTemp double);
+    
+    INSERT INTO OutputStream
+    SELECT roomNo, avg(temp) AS avgTemp
+    FROM TempStream
+    GROUP BY roomNo;
     ```
 
-## Transform data using Plugins
+## Transform Data Using Plugins
 
 The stream processor offers a variety of options to carry out data transformations via in-built plugins. The following table describes the complete list of plugins that provide data transformation functionality.
```

Comment:
By stream processor, do you mean stream worker?

Reply:
I think stream processor, 

Stream Processor

This allows events to be collected, generated, dropped and expired anytime by altering the event format by adding one or more attributes to it based on the given input parameters.